Display: Allow unthrottle to skip only flipping

unknownbrackets opened the pull request:
Before, it either flipped continuously, or forced frameskip on.  This makes it so you can still draw frames, but skip actual flips.

This is useful when games draw things only in a single frame and reuse later.  It's also useful when measuring speed improvements if you already get 100% speed on a device.

This resets the previous FrameskipUnthrottle setting.  That said, this may be a good thing because some users changed it when vsync/postshaders were not interacting as well with it, and I think those problems are fixed.

-[Unknown]

unknownbrackets committed:
Display: Allow unthrottle to skip only flipping.

Before, it either flipped continuously, or forced frameskip on.  This
makes it so you can still draw frames, but skip actual flips.

This is useful when games draw things only in a single frame and reuse
later.  It's also useful when measuring speed improvements if you already
get 100% speed on a device.

hrydgard commented:
Yay, I've wanted this for a long time. On a fence whether to merge it for 1.10...

hrydgard commented:
Gonna need a little rebase here, sorry :)

unknownbrackets committed:
EditorConfig: Separate indent style for libretro.

Three spaces is weird, but better to be consistent and have the config
specify what's being used.

hrydgard commented:
There was another conflict (caused by my latest changes) so I merged locally. Closing.

unknownbrackets commented:
We may even want to consider exposing this in the UI or making it default to SKIP_FLIP when frameskip is otherwise off.

-[Unknown]
